README.md

```diff
@@ -1,146 +1,219 @@
 # witnet-ethereum-bridge [![](https://travis-ci.com/witnet/witnet-ethereum-bridge.svg?branch=master)](https://travis-ci.com/witnet/witnet-ethereum-bridge)
 
-`witnet-ethereum-bridge` is an open source implementation of a bridge
-from Ethereum to Witnet. This repository provides several contracts:
+`witnet/witnet-ethereum-bridge` is an open source implementation of an API that enable Solidity smart contract developers to harness the full power of the [**Witnet Decentralized Oracle Network**](https://docs.witnet.io/overview/concepts/).
 
-- The `WitnetRequestBoard` (WRB), which includes all the needed functionality to relay data requests and their results from Ethereum to Witnet and the other way round.
-- `WitnetRequestBoardProxy`, that routes Witnet data requests from smart contracts to the appropriate `WitnetRequestBoard` controller.
-- `UsingWitnet`, an inheritable client contract that injects methods for interacting with the WRB in the most convenient way.
+This repository provides several deployable contracts:
 
+- `WitnetProxy`, that routes Witnet data requests to a currently active `WitnetRequestBoard` implementation.
+- `WitnetRequest`, used as a means to encapsulate CBOR-encoded Witnet RADON scripts.
+- `WitnetRequestBoard` (WRB), which implements all required functionality to relay Witnet data requests (i.e. encapsulated [Witnet RADON scripts](https://docs.witnet.io/protocol/data-requests/overview/)) from Ethereum to the Witnet mainnet, as well as to relay Witnet-solved results back to Ethereum.
+- `UsingWitnet`, an inheritable abstract contract that injects methods for interacting with the WRB in the most convenient way.
```

Comment:
```suggestion
- `UsingWitnet`, an inheritable abstract contract that injects methods for conveniently interacting with the WRB.
```

Maybe the `UsingWitnet` should not be on this list as by itself is not deployeable (abstract contract).

Reply:
```
- `WitnetRequestBoard` (WRB), which implements all required functionality to relay Witnet data requests (i.e. encapsulated [Witnet RADON scripts](https://docs.witnet.io/protocol/data-requests/overview/)) from Ethereum to the Witnet mainnet, as well as to relay Witnet-solved results back to Ethereum.

The repository also provides:

- `UsingWitnet`, an inheritable abstract contract that injects methods for conveniently interacting with the WRB.
```